stratosphere/ncm/source/impl/lr_registered_data.hpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2019 Adubbz
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include <switch.h>
+#include <stratosphere.hpp>
+
+namespace ams::lr::impl {
+
+    template<typename Key, typename Value, size_t NumEntries>
+    class RegisteredData {
+        NON_COPYABLE(RegisteredData);
+        NON_MOVEABLE(RegisteredData);
+        private:
+            struct Entry {
+                Value value;
+                ncm::ProgramId owner_id;
+                Key key;
+                bool is_valid;
+            };
+        private:
+            Entry entries[NumEntries];
+            size_t soft_entry_limit;
```

Comment:
`soft_entry_limit` -- is there any reason this isn't called `capacity` or `size` or `num_entries`?